rundeckapp/grails-app/domain/rundeck/ScheduledExecution.groovy

```diff
@@ -562,33 +556,65 @@ class ScheduledExecution extends ExecutionContext implements EmbeddedJsonData {
             def nots=[]
             data.notification.keySet().findAll{it.startsWith('on')}.each{ name->
-                if(data.notification[name]){
+                def notifications = data.notification[name]
+                if(notifications){
                         //support for built-in notification types
-                    ['urls','email'].each{ subkey->
-                        if(data.notification[name][subkey]){
-                            Map notificationData = [
-                                    format    : data.notification[name]["format"],
-                                    httpMethod: data.notification[name]["httpMethod"],
-                                    (subkey)  : data.notification[name][subkey]
-                            ]
-
-                            nots << Notification.fromMap(name, notificationData)
+                    if(notifications instanceof ArrayList){
+                        //TODO: iterate through every notification
+                        notifications.each { notifMap ->
+                            ['urls','email'].each{ subkey->
+                                if(notifMap[subkey]){
+                                    Map notificationData = [
+                                            format    : notifMap["format"],
+                                            httpMethod: notifMap["httpMethod"],
+                                            (subkey)  : notifMap[subkey]
+                                    ]
+                                    nots << Notification.fromMap(name, notificationData)
+                                }
+                            }
+                            if(notifMap['plugin']){
+                                def pluginElement=notifMap['plugin']
+                                def plugins=[]
+                                if(pluginElement instanceof Map){
+                                    plugins=[pluginElement]
+                                }else if(pluginElement instanceof Collection){
+                                    plugins= pluginElement
+                                }
+                                plugins.each{ plugin->
+                                    def n=Notification.fromMap(name, plugin)
+                                    if(n){
+                                        nots << n
+                                    }
+                                }
+                            }
                         }
-                    }
-                    if(data.notification[name]['plugin']){
-                        def pluginElement=data.notification[name]['plugin']
-                        def plugins=[]
-                        if(pluginElement instanceof Map){
-                            plugins=[pluginElement]
-                        }else if(pluginElement instanceof Collection){
-                            plugins= pluginElement
-                        }else{
-
+                    }else {
```

Comment:
This was left to keep compatibility with yaml notifications of previous versions